featuretools/primitives/transform_primitive.py

```diff
@@ -336,6 +337,30 @@ class Weekday(DatetimeUnitBasePrimitive):
     name = "weekday"
 
 
+class NumCharacter(TransformPrimitive):
+    """
+    Return the characters in a given string.
+    """
+    name = 'characters'
+    input_types = [Text]
+    return_type = Numeric
+
+    def get_function(self):
+        return lambda array: pd.Series([len(x) for x in array])
+
+
+class NumWords(TransformPrimitive):
+    """
+    Returns the words in a given string by counting the spaces.
+    """
+    name = 'numwords'
+    input_types = [Text]
+    return_type = Numeric
+
+    def get_function(self):
+        return lambda array: pd.Series([len(x.split(" ")) for x in array])
```

Comment:
similar to above, you can do 

```python

def get_function(self):
    return lambda array: array.str.split(" ").str.len()
```

Reply:
actually, this is probably better. 

```python
def get_function(self):
    return lambda array: array.str.count(" ")  + 1
```

easier to read and might be [up to 25% faster](https://stackoverflow.com/a/37335656)

Reply:
Those aren't quite equivalent - what if there's some leading or trailing whitespace, or if there's more than one space between characters?

Reply:
I think both pieces of code handle white space and more than one space the same way because they look for just " ".  To do the multiple spaces, the only way I can think of is a regex and I rather keep it simple for now. However, for the trailing or leading white space we could do

```python

def get_function(self):
    return lambda array: array.str.strip().str.count(" ") + 1

```

Reply:
Multiple spaces could be taken care of in the str.split(" ") case since they'll show up as empty strings. Could just remove with
```
new_list = [x for x in split_list if x != '']
```
For now, it has been changed to `str.count(" ") + 1`